packages/core/src/codewhisperer/util/commonUtil.ts

```diff
@@ -30,6 +30,17 @@ export function asyncCallWithTimeout<T>(asyncPromise: Promise<T>, message: strin
     })
 }
 
+export function asyncCallWithTimeoutSwallowErrors<T>(asyncPromise: Promise<T>, timeLimit: number): Promise<T> {
+    let timeoutHandle: NodeJS.Timeout
+    const timeoutPromise = new Promise((resolve) => {
+        timeoutHandle = setTimeout(() => resolve(undefined), timeLimit)
+    })
+    return Promise.race([asyncPromise, timeoutPromise]).then((result) => {
+        clearTimeout(timeoutHandle)
+        return result as T
+    })
+}
```

Comment:
don't quite understand why we need a different version of `asyncCallWithTimeout` with the only difference where we don't handle reject here. Can't we use `asyncCallWithTimeout`?

Reply:
at least they seems to have pretty much  the same functionality?

Reply:
No it throws. `asyncCallWithTimeout` throws and when we `await Promise.all([promiseA, promiseB)]`, if promiseA throws then the entire promise throws. So we need A and B to not throw instead return undefined